src/mca/ptl/base/ptl_base_fns.c

```diff
@@ -1458,33 +1469,47 @@ char **pmix_ptl_base_split_and_resolve(char **orig_str, char *name)
 
         /* Go through all interfaces and see if we can find a match */
         for (if_index = pmix_ifbegin(); if_index >= 0; if_index = pmix_ifnext(if_index)) {
-            pmix_ifindextoaddr(if_index, (struct sockaddr *) &if_inaddr, sizeof(if_inaddr));
-            if (pmix_net_samenetwork(&argv_inaddr, &if_inaddr, argv_prefix)) {
-                break;
+            pmix_ifindextoaddr(if_index,
+                               (struct sockaddr*) &if_inaddr,
+                               sizeof(if_inaddr));
+            if (pmix_net_samenetwork((struct sockaddr*) &argv_inaddr,
+                                     (struct sockaddr*) &if_inaddr,
+                                     argv_prefix)) {
+                /* We found a match. If it's not already in the interfaces array,
+                   add it. If it's already in the array, treat it as a match */
+                match_count = match_count + 1;
+                pmix_ifindextoname(if_index, if_name, sizeof(if_name));
+                for (n = 0; n < interface_count; n++) {
+                    if (0 == strcmp(if_name, interfaces[n])) {
+                        break;
+                    }
+                }
+                if (n == interface_count) {
+                    pmix_output_verbose(20,
+                                        pmix_ptl_base_framework.framework_output,
+                                        "ptl:tool: Found match: %s (%s)",
+                                        pmix_net_get_hostname((struct sockaddr*) &if_inaddr),
+                                        if_name);
+                    pmix_argv_append(&interface_count, &interfaces, if_name);
+                }
             }
+
         }
         /* If we didn't find a match, keep trying */
-        if (if_index < 0) {
+        if (0 == match_count) {
             pmix_show_help("help-ptl-base.txt", "invalid if_inexclude", true, name, tmp,
                            "Did not find interface matching this subnet");
             free(tmp);
             continue;
         }
-
-        /* We found a match; get the name and replace it in the
-           argv */
-        pmix_ifindextoname(if_index, if_name, sizeof(if_name));
-        pmix_output_verbose(20, pmix_ptl_base_framework.framework_output,
-                            "ptl:tool: Found match: %s (%s)",
-                            pmix_net_get_hostname((struct sockaddr *) &if_inaddr), if_name);
-        argv[save++] = strdup(if_name);
-        free(tmp);
     }
 
-    /* The list may have been compressed if there were invalid
-       entries, so ensure we end it with a NULL entry */
-    argv[save] = NULL;
+    /* Mark the end of the interface name array with NULL */
+    if (NULL != interfaces) {
```

Comment:
You don't need nor want to add a NULL to the end of this array. The `pmix_argv_append` function ensures that there is a NULL entry at the end of the array.